src/components/DAO/ExpandDAO/ChooseVotingStrategies/index.tsx

```diff
@@ -18,93 +22,92 @@ const ChooseVotingStrategies: FunctionComponent<{
 	onStrategyAdd: (strategy: BuiltVotingStrategy) => void
 	onStrategyRemove: (index: number) => void
 	onSubmit: () => void
-}> = ({gnosisAddress, strategies, transactions, onStrategyAdd, onStrategyRemove, onSubmit}) => {
+}> = ({gnosisAddress, strategies, onStrategyAdd, onStrategyRemove, onSubmit}) => {
+	const {signer} = useContext(EthersContext)
 	const [addStrategyModalOpened, setAddStrategyModalOpened] = useState<VotingStrategy | null>(null)
-	const [removeStrategyModalOpened, setRemoveStrategyModalOpened] = useState<number | null>(null)
 
-	const handleSubmitVotingStrategy = (votingStrategy: BuiltVotingStrategy) => {
-		onStrategyAdd(votingStrategy)
-		setAddStrategyModalOpened(null)
+	const handleSubmitVotingStrategy = async (
+		strategy: VotingStrategy,
+		{erc20TokenAddress, quorumThreshold, delay, votingPeriod}: VotingStrategyFormValues
+	) => {
+		if (strategy === "linearVoting") {
```

Comment:
I don't see any point in having 6 modals with 50% of the same content and form fields. I would rather handle that via hooks if the validation and API calling logic will differ.
We gonna revisit this architecture anyway once the skills tree design will be rolled out, and if the form should become more complicated or differ significantly for different strategies - then for me it would make sense to split them. For now, your suggestion is premature optimization, which might not make any sense due to the upcoming revision.
WDYT to keeping it same right now and separating to different components in future, if such a need will occur?

Reply:
My opinion is completely opposite.
Duplication is better than bad abstraction.
We have only 1 strategy now and we don't have an idea, what API and params the other strategies would have.
Premature optimization is what you do now - you are trying to create abstraction before we have things to abstract.
We can add several strategies (and it can take quite a long) and if we see that there's a lot of duplication in our code and we can abstract it out, we will do it. This is the easier way, I think

Reply:
> We have only 1 strategy now and we don't have an idea, what API and params the other strategies would have.

Actually - we do have clue at least about params and inputs, check the following designs:

https://xd.adobe.com/view/682820cf-2729-46f3-71cc-6cb9a332fc09-1e19/screen/57ccc050-11b4-487f-a605-02953b11e0ec/specs/
https://xd.adobe.com/view/682820cf-2729-46f3-71cc-6cb9a332fc09-1e19/screen/42f99e27-37de-40c7-bd00-18cb45e4928d/specs/
https://xd.adobe.com/view/682820cf-2729-46f3-71cc-6cb9a332fc09-1e19/screen/b118604e-c505-43cd-be34-b4baed3c11da/specs/
https://xd.adobe.com/view/682820cf-2729-46f3-71cc-6cb9a332fc09-1e19/screen/a274c88c-28d9-47f4-b124-cca5ead24341/specs/
https://xd.adobe.com/view/682820cf-2729-46f3-71cc-6cb9a332fc09-1e19/screen/aa6c8304-73bf-4217-a5a1-84b2b36f694c/specs/
https://xd.adobe.com/view/682820cf-2729-46f3-71cc-6cb9a332fc09-1e19/screen/6145f9dc-5cca-4ec4-a3e4-d920dd3f19cf/specs/

API of course might differ, and final implementation - as well. But coming up from "what we have right now" - those modals and inputs there are nearly the same. Bad abstraction is worse than duplication, but this is not a bad abstraction, coz it is based on the current state of things, rather than trying to predict what it could be. 
The abstraction makes it extremely small, while duplication will increase the code base for this small modal part ~5 times.

Again, seems like we need some additional opinion on this topic

Reply:
There will be many strategies that we can't predict at the moment. However there are many that you can see do share the majority of their inputs. I find an abstraction for the current state of things fine for now. We can always revisit :)